powersimdata/input/electrified_demand_input.py

```diff
@@ -0,0 +1,43 @@
+from powersimdata.input.profile_input import ProfileInput
+
+
+def get_profile_version(_fs, grid_model, kind, end_use, tech):
+    _fs = _fs.makedirs(f"raw/{grid_model}/{kind}", recreate=True)
+    base_name = f"{end_use}_{tech}_"
+    matching = [f for f in _fs.listdir(".") if base_name in f]
+
+    return [f.replace(base_name, "").replace(".csv", "") for f in matching]
```

Comment:
Note: I changed `lstrip` and `rstrip` to `replace` here because it turns out those strip any of the given characters starting from one side until one doesn't match, so it could lead to unpredictable results.

Reply:
What if we have `'demand'` or `'solar'` or whatever elsewhere within the string, it seems that will get replaced as well. Should we do this via regex instead? E.g.
```python
>>> import re
>>> s = "ab_cd_ef"
>>> match = re.search("^(?P<prefix>ab_)?(?P<main>[^_]+)(?P<suffix>_ef)?$", s)
>>> match.group('prefix')
'ab_'
>>> match.group('main')
'cd'
>>> match.group('suffix')
'_ef'
```

Reply:
String parsing methods are definitely clearer, but have the potential for weird side effects if the strings are not constructed in the way that we want.

Reply:
With this implementation, `kind` is either "building" or "transportation". So the profiles in each of those folders will have the structure `[end_use]_[tech]_[version].csv`.

Reply:
Agree. I will defer to @jon-hagg's decision at this point.

Reply:
I think since we are responsible for profile names and version, and in this case replacing the entire prefix up to the version, it's unlikely we run into issues. E.g. the version should hopefully not contain ".csv" or "res_cooking_standard_heat_pump". I do like regex, but since the naming is so consistent we probably don't need it at this point?

Reply:
It does operate on the local folder. I guess we should document the expected structure for custom profiles if we don't have that yet. My assumption is that it's not _supposed_ to work for arbitrary names, just custom versions.

Reply:
I trust your judgement. It's not like we have a ton of users out there adding a bunch of weirdly named files yet.